scripts/taskprolog.sh

```diff
@@ -2,6 +2,8 @@
 if [ -n "$_VENODELIST" ]; then
     declare -a VE_NODE_IDS
     VE_NODE_IDS=($_VENODELIST)
-    echo "export VE_NODE_NUMBER=${VE_NODE_IDS[$SLURM_LOCALID]}"
+    NB_VE=${#VE_NODE_IDS[@]}
+    LOCALID=$(( ${SLURM_LOCALID} % ${NB_VE} ))
+    [ -z "${SLURMD_TRES_BIND}" ] && echo "export VE_NODE_NUMBER=${VE_NODE_IDS[$LOCALID]}"
```

Comment:
Are you sure VE_NODE_NUMBER is set inside _set_env_vars() for every step when SLURMD_TRES_BIND is set? My impression was that _set_env() is not (always) being called for each slurmstep.